nym-vpn-app/src-tauri/src/grpc/client.rs

```diff
@@ -59,13 +60,31 @@ pub enum VpndError {
 }
 
 #[derive(Debug, Default, Clone)]
-pub struct GrpcClient(Transport);
+pub struct GrpcClient {
+    transport: Transport,
+    user_agent: UserAgent,
+}
 
 impl GrpcClient {
     #[instrument(skip_all)]
-    pub fn new(config: &AppConfig, cli: &Cli) -> Self {
-        let client = GrpcClient(Transport::from((config, cli)));
-        match &client.0 {
+    pub fn new(config: &AppConfig, cli: &Cli, pkg: &PackageInfo) -> Self {
+        let git_commit = crate::build_info()
+            .version_control
+            .as_ref()
+            .and_then(|vc| vc.git())
+            .map(|g| g.commit_short_id.clone())
+            .unwrap_or_default();
+
+        let client = GrpcClient {
+            transport: Transport::from((config, cli)),
+            user_agent: UserAgent {
+                application: pkg.name.clone(),
+                version: pkg.version.to_string(),
+                platform: format!("{}-{}", OS, ARCH),
```

Comment:
Please see the ticket and the surrounding discussion regarding the format. Also I don't think `OS` will give you specific enough info, what is the resulting string?

Reply:
Ah, yes the OS version would be useful

Reply:
Check the PR for nym-vpnc which contains an example for what it's set to. Note the field separators for the platform. There is a bit more material in the Jira ticket about the choice of format

```rust
UserAgent {
    application: "nym-vpnc",
    version: "0.2.2-dev (0.2.2-dev)",
    platform: "Ubuntu; Linux 23.10 Ubuntu; x86_64",
    git_commit: "6dde4bd8bda00cce89bc2a0c75dac267d3451882 (ba519d1cefb65d48fe6997ec303932fbb31fe4c6)",
}
```